diff --git a/vllm/utils/network_utils.py b/vllm/utils/network_utils.py
index 6b940c92d..230f96c3d 100644
--- a/vllm/utils/network_utils.py
+++ b/vllm/utils/network_utils.py
@@ -146,6 +146,7 @@ def get_open_zmq_ipc_path() -> str:
 def get_open_zmq_inproc_path() -> str:
     return f"inproc://{uuid4()}"

+_next_port: int | None = None

 def get_open_port() -> int:
     """
@@ -159,11 +160,15 @@ def get_open_port() -> int:
+    global _next_port
     if "VLLM_DP_MASTER_PORT" in os.environ:
         dp_master_port = envs.VLLM_DP_MASTER_PORT
         reserved_port_range = range(dp_master_port, dp_master_port + 10)
         while True:
-            candidate_port = _get_open_port()
+            candidate_port = _get_open_port(_next_port)
+            _next_port = candidate_port + 1
             if candidate_port not in reserved_port_range:
                 return candidate_port
-    return _get_open_port()
+    port = _get_open_port(_next_port)
+    _next_port = port + 1
+    return port


 def get_open_ports_list(count: int = 5) -> list[int]:
