diff --git a/sglang/multimodal_gen/configs/pipeline_configs/zimage.py b/sglang/multimodal_gen/configs/pipeline_configs/zimage.py
index 68f00a667..97950e7f2 100644
--- a/sglang/multimodal_gen/configs/pipeline_configs/zimage.py
+++ b/sglang/multimodal_gen/configs/pipeline_configs/zimage.py
@@ -6,10 +6,8 @@ import torch
 
 from sglang.multimodal_gen.configs.models import DiTConfig, EncoderConfig, VAEConfig
 from sglang.multimodal_gen.configs.models.dits.zimage import ZImageDitConfig
-from sglang.multimodal_gen.configs.models.encoders import (
-    BaseEncoderOutput,
-    TextEncoderConfig,
-)
+from sglang.multimodal_gen.configs.models.encoders import BaseEncoderOutput
+from sglang.multimodal_gen.configs.models.encoders.qwen3 import Qwen3TextConfig
 from sglang.multimodal_gen.configs.models.vaes.flux import FluxVAEConfig
 from sglang.multimodal_gen.configs.pipeline_configs.base import (
     ImagePipelineConfig,
@@ -44,7 +42,7 @@ class ZImagePipelineConfig(ImagePipelineConfig):
     dit_config: DiTConfig = field(default_factory=ZImageDitConfig)
     vae_config: VAEConfig = field(default_factory=FluxVAEConfig)
     text_encoder_configs: tuple[EncoderConfig, ...] = field(
-        default_factory=lambda: (TextEncoderConfig(),)
+        default_factory=lambda: (Qwen3TextConfig(),)
     )
 
     preprocess_text_funcs: tuple[Callable, ...] = field(
